shared/src/main/java/net/blay09/mods/waystones/core/WarpMode.java

```diff
@@ -17,6 +17,7 @@ public enum WarpMode {
     WAYSTONE_TO_WAYSTONE(() -> WaystonesConfig.getActive().xpCost.waystoneXpCostMultiplier, WarpMode::waystoneIsActivated, false),
     SHARESTONE_TO_SHARESTONE(() -> WaystonesConfig.getActive().xpCost.sharestoneXpCostMultiplier, WarpMode::sharestonesOnly, false),
     WARP_PLATE(() -> WaystonesConfig.getActive().xpCost.warpPlateXpCostMultiplier, WarpMode::warpPlatesOnly, false),
+    WARP_PLATE_CONSUMES(() -> WaystonesConfig.getActive().xpCost.warpPlateXpCostMultiplier, WarpMode::warpPlatesOnly, true),
```

Comment:
I would prefer not adding another warp mode as I intend to move away from using this class at all - since the warp plate is in charge of the teleport, it can take care of removing the item on its own

Reply:
The `WaystoneTeleportContext` was added fairly recently and eventually I would see the different warp mode specialties within fields of that class instead.